### DISCUSSION TITLE
[Chapter 3] Q & A

### DISCUSSION THREAD

**Discussion by JoisFe** [Study-2-Modern-Java-In-Action/Study-2-Modern-Java-in-Action#35] [CHAPTER 3 람다 표현식 - Q&A]
# [Chapter 3] Q & A

### Q1) 추상 메서드 시그니처가 곧 람다 표현식의 시그니처
메서드 시그니처 가 무엇? (어떠한 것이 시그니처?)

- throws 또한 시그니처?

### Q3) 

``` java
        List<String> list = new ArrayList<>();

        Predicate<String> p = s -> list.add(s);  // (1)
        
        Consumer<String> b = s -> list.add(s); // (2)
```

- (1), (2) 중 컴파일 에러가 나는 것은 ??

--- 
### Q4) 메서드 참조의 세 가지 유형 중 `기존 객체의 인스턴스 메서드 참조`의 설명 중 `비공개 헬퍼 메서드를 정의한 상황에서 유용하게 사용할 수 있다.` 라는 글이 있는데 여기서 비공개 헬퍼 메서드가 무엇인가요?

**Comment by corock** [comment]
A1. `throws` 는 메서드를 호출하는 쪽에서 예외처리를 하도록 책임 전가를 하는 녀석이라 호출하는 주체에 따라 달라질 수 있어 _signature_ 가 아닐까 싶습니다.

A2. `@FunctionalInterface` 를 추가한 인터페이스는 추상 메서드 2개 이상 선언하는 순간 컴파일 오류가 뜹니다.

**Reply by JoisFe** [reply]
제가 Q&A 제목을 이상하게 지었내요
처음에 `컴파일러가 컨텍스트를 통해 타입을 추론할 수 있다 -> 모든 경우 다 그럴까?` 컨셉의 의문이 들었는데 타입을 추론 못하는 경우를 만들기 쉽지 않더라구요..
